### PR TITLE
doc: KFP 2021 roadmap

### PR DESCRIPTION
**Pull request by Bobgy**
**Description of your changes:**
Update KFP 2021 roadmap

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -1,10 +1,101 @@
...
+* Alpha release in October
+* Beta/Stable release timelines TBD
+
+#### Planned Features in KFP v2
+
+* KFP v2 DSL (TBD)
+
+* Use [the pipeline spec](https://github.com/kubeflow/pipelines/blob/master/api/v2alpha1/pipeline_spec.proto) as pipeline description data structure. The new spec is argo workflow agnostic and can be a shared common format for different underlying engines.
+
+* KFP v2 DAG UI
```
Comment (zijianjoy) [inline ROADMAP.md:78]: Do we also mention storing and reading pipeline run status to MLMD?

File: ROADMAP.md [hunk 2]
```diff
@@ -1,10 +1,101 @@
...
+
+#### KFP v2 Goals
+
+* Data Management:
+  * build first class support for metadata -- recording, presentation and orchestration.
+  * making it easy to keep track of all the data produced by machine learning pipelines and how it was computed.
+* KFP native (and argo agnostic) spec and status: define a clear interface for KFP, so that other systems can understand KFP pipeline spec and status in KFP semantics.
+* Gaining more control over KFP runtime behavior: give the KFP system more control over the exact runtime behavior. This wasn’t a goal initially coming from use cases. However, the more we innovate on Data Management and KFP native spec/status, the clearer that other workflow systems become limitations to how we may implement new KFP features on our own. Therefore, re-architecturing KFP to let us get more control of runtime behavior is ideal for achieving KFP’s long term goals.
+* Be backward compatible with KFP v1: for existing features, we want to keep them as backward compatible as possible to ease upgrade.
+
```
Comment (james-jwu) [inline ROADMAP.md:65]: v2 also sets up KFP for supporting new features, such as metadata base caching, new control flow features such as fan in.

File: ROADMAP.md [hunk 3]
```diff
@@ -1,10 +1,101 @@
-# Kubeflow Pipelines 2019 Roadmap
+# Kubeflow Pipelines Roadmap
```
Comment (james-jwu) [inline ROADMAP.md:1]: "2021 Roadmap"?

File: ROADMAP.md (hunk 2 above)
Comment (Bobgy) [reply ROADMAP.md:65]: thanks, added

File: ROADMAP.md (hunk 1 above)
Comment (Bobgy) [reply ROADMAP.md:78]: that feels like an implementation detail to me.
I included that in user facing item: "KFP UI in KFP semantics".

Comment (zijianjoy) [reply ROADMAP.md:81]: Sounds good to me, thank you!

**Comment by zijianjoy** [comment]
/lgtm

Thank you Yuan! Feel free to merge or wait for intended audiences to approve.

**Comment by Bobgy** [comment]
Thanks, I think we can keep improving in separate PRs if needed. It's good getting the roadmap added.
/approve

**Pull request merged**